### PR TITLE
Lock node-version for dev & ci, add eslint

### PR DESCRIPTION
**Pull request by rheaditi**
* Use node 13.x for local development & ci
* Add eslint from typescript-action
* Add lint & format into PR check

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+module.exports = {
```
Comment (rheaditi) [inline .eslintrc.js:1]: this file is taken directly from [typescript-action's version](https://github.com/actions/typescript-action/blob/9cdc5b3/.eslintrc.json) with few rules tweaked (semi).

File: src/main.ts [hunk 2]
```diff
@@ -3,43 +3,43 @@ import * as github from '@actions/github';
 import { PullsUpdateParams, IssuesCreateCommentParams } from '@octokit/rest';
 
 import {
-  getJIRAClient,
```
Comment (rheaditi) [inline src/main.ts:6]: arranged in ascending order here

File: .eslintrc.js (hunk 1 above)
Comment (rajanand02) [reply .eslintrc.js:1]: :clap: :medal_military:

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: "test-typescript-action"
```
Comment (rajanand02) [inline .github/workflows/checks.yml:1]: Please rename this. this will show up in the actions log

File: .github/workflows/pr-branch-checks.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      # ignore node_modules since dev/fresh ci deps installed.
+      run: |
+        git diff --exit-code --stat -- . ':!node_modules' \
+        || (echo "##[error] found changed files after build. please 'npm run build'" \
+                  "and check in all changes" \
+            && exit 1)
+
+    - name: "archive lib when diff"
+      uses: actions/upload-artifact@v1
+      if: failure()
```
Comment (rheaditi) [inline .github/workflows/pr-branch-checks.yml:38]: only run if previous step fails

**Pull request merged**